### PR TITLE
Remove construct dependency entirely.

### PR DESCRIPTION
**Pull request by paulprozesky**
@brickZA - please to check this and confirm.

### CODE REVIEW

File: src/bitfield.py [hunk 1]
```diff
@@ -1,18 +1,16 @@
...
 
 LOGGER = logging.getLogger(__name__)
 
 
 class Bitfield(object):
-    """Wraps contruct's BitStruct class. It describes a chunk of memory that
-    consists of a number of fields.
     """
-    def __init__(self, name, width, fields=None):
+    Describes a chunk of memory that consists of a number of Field.
```
Comment (brickZA) [inline src/bitfield.py]: number of Field_s_ ? 

status: Done

File: src/bitfield.py [hunk 2]
```diff
@@ -93,38 +72,40 @@ def fields_string_get(self):
...
-        Initialise a field object.
-        @param name  The name of the field
-        @param numtype  A numerical description of the type - 0 is unsigned,
-            1 is signed 2's comp and 2 is boolean
-        @param width  The width of the field, in bits
-        @param binary_pt  The binary point position, in bits
-        @param lsb_offset  The offset in the memory field, in bits - -1 means it hasn't been set yet.
+        Initialise a Field object.
+        :param name: The name of the field
+        :param numtype: A numerical description of the type - 0 is unsigned, 1 is signed 2's comp and 2 is boolean
```
Comment (brickZA) [inline src/bitfield.py]: Perhaps break up the long line?

Status: Done

File: src/memory.py [hunk 3]
```diff
@@ -34,6 +35,56 @@ def bin2fp(raw_word, bitwidth, bin_pt, signed):
...
+    described by the params.
+    Provides the same output as a Xilinx block in Simulink would if you cast it to an unsigned int.
+    """
+    _format = '%s%i.%i' % ('fix' if signed else 'ufix', bitwidth, bin_pt)
+    LOGGER.debug('Converting %f to %s' % (num, _format))
+    if bin_pt >= bitwidth:
+        LOGGER.error('    Cannot have bin_pt >= bitwidth')
+        raise ValueError('Cannot have bin_pt >= bitwidth')
+    if bin_pt < 0:
+        LOGGER.error('    bin_pt < 0 makes no sense')
```
Comment (brickZA) [inline src/memory.py]: Probably no reason to log if you are also raising an error?

status: Done

**Comment by paulprozesky** [comment]
Changes pushed to address comments above.

File: src/register.py [hunk 4]
```diff
@@ -97,48 +88,53 @@ def write_int(self, uintvalue, blindwrite=False, word_offset=0):
...
     def _write_common(self, **kwargs):
-        # TODO - make read before write optional - otherwise just write the whole word based on last-known values
         """
-        :param kwargs:
+        Form the dictionary of values that must be written
+        :param kwargs: the field names and values to write
         :return:
         """
-        if len(kwargs) == 0:
+        if len(kwargs.keys()) == 0:
```
Comment (brickZA) [inline src/register.py]: Should be no need to make this change -- len() on a dict should work like you want it to.

Comment (paulprozesky) [inline src/register.py]: Okay, so changed.

**Pull request merged**